tests/integration/test_fim/test_files/test_whodata_policy_change/test_whodata_policy_change.py

```diff
@@ -0,0 +1,174 @@
+def test_whodata_policy_change(configuration, metadata, set_wazuh_configuration, create_monitored_folders_module,
+                               configure_local_internal_options_function, restart_syscheck_function,
+                               wait_fim_start_function):
+    '''
+    description: Check if the 'wazuh-syscheckd' is monitoring a in whodata mode in Windows, and the Audit Policies are
+                 changed, the monitoring changes to realtime and works on the monitored files.
+
+    wazuh_min_version: 4.5.0
+
+    tier: 1
+
+    parameters:
+        - configuration:
+            type: dict
+            brief: Configuration values for ossec.conf.
+        - metadata:
+            type: dict
+            brief: Test case data.
+        - create_monitored_folders_module:
+            type: fixture
+            brief: Create the folders that will be monitored, delete them after test.
+        - set_wazuh_configuration:
+            type: fixture
+            brief: Set ossec.conf configuration.
+        - configure_local_internal_options_function:
+            type: fixture
+            brief: Set local_internal_options configuration.
+        - restart_syscheck_function:
+            type: fixture
+            brief: restart syscheckd daemon, and truncate the ossec.log.
+        - wait_for_fim_start_function:
+            type: fixture
+            brief: check that the starting fim scan is detected.
+
+    assertions:
+        - Verify the SACL for the monitored files is configured
+        - Verify Whodata monitoring has started
+        - Verify that the event 4719 event is detected and changes monitoring to real-time
+        - Verify the monitoring mode changes to real-time
+        - Verify monitoring in real-time works correctly for the monitored files.
+
+    input_description:
+        - The file 'cases_whodata_policy_change.yaml' provides the test cases and specific configuration.
+        - The file 'configuration_whodata_policy_change.yaml' provides the configuration template to be used.
+
+    expected_output:
+        - fr".*win_whodata.*The SACL of '({file})' will be configured"
+        - r'.*win_whodata.*(Event 4719).*Switching directories to realtime'
+        - fr".*set_whodata_mode_changes.*The '({file})' directory starts to be monitored in real-time mode."
+        - r'.*Sending FIM event: (.+)$' ('added', 'modified', and 'deleted' events)
+
+    tags:
+        - whodata
+    '''
+    wazuh_log_monitor = FileMonitor(LOG_FILE_PATH)
+
+    # Check it is being monitored in whodata
+    evm.detect_windows_sacl_configured(wazuh_log_monitor, '.*')
+    # Check Whodata engine has started
+    evm.detect_whodata_start(wazuh_log_monitor)
+
+    # Change policies
+    if metadata['check_event']:
+        # Wait to allow thread_checker to be executed twice so Event 4719 detection starts.
+        time.sleep(6)
```

Comment:
Change this hardcoded value into a metadata value or something similar

Reply:
Done in https://github.com/wazuh/wazuh-qa/pull/3627/commits/0d431cc2ea9340d5f7d0aef41bc14acd57a26ff0